integration-tests/implementationGuides.test.ts

```diff
@@ -93,14 +128,97 @@ describe('Implementation Guides - US Core', () => {
+                },
+            ],
+        },
+    };
+
+    describe('Updating patient', () => {
+        let patientId = '';
+        beforeAll(async () => {
+            const patient = getRandomPatientWithEthnicityAndRace();
+            const { data } = await client.post('Patient', patient);
+            patientId = data.id;
+        });
+
+        test('updating Patient with valid resource', async () => {
+            const patient = getRandomPatientWithEthnicityAndRace();
+            patient.id = patientId;
+
+            // const expectedPatient: any = cloneDeep(patient);
+            await expect(client.put(`Patient/${patientId}`, patient)).resolves.toMatchObject({
+                status: 200,
+                data: patient,
+            });
+        });
+
+        test('updating Patient with invalid resource: no text field', async () => {
+            const patient = getRandomPatientWithEthnicityAndRace();
+            patient.id = patientId;
+
+            // Remove text field
+            delete patient.extension[0].extension[1];
+            delete patient.extension[1].extension[1];
+
+            await expect(client.put(`Patient/${patientId}`, patient)).rejects.toMatchObject({
+                response: noTextFieldErrorResponse,
+            });
+        });
+    });
+
+    test('creating valid US Core patient', async () => {
+        const patient = getRandomPatientWithEthnicityAndRace();
+
+        const expectedPatient: any = cloneDeep(patient);
+        delete expectedPatient.id;
+        await expect(client.post('Patient', patient)).resolves.toMatchObject({
+            status: 201,
+            data: expectedPatient,
+        });
+    });
+
+    test('creating invalid US Core patient: no text field', async () => {
+        const patient = getRandomPatientWithEthnicityAndRace();
+        // Remove text field
+        delete patient.extension[0].extension[1];
+        delete patient.extension[1].extension[1];
+        await expect(client.post('Patient', patient)).rejects.toMatchObject({
+            response: noTextFieldErrorResponse,
+        });
+    });
+
+    test('query using search parameters', async () => {
+        const patient = getRandomPatientWithEthnicityAndRace();
 
-        const testPatient: ReturnType<typeof randomPatient> = (await client.post('Patient', usCorePatient)).data;
+        const testPatient: ReturnType<typeof randomPatient> = (await client.post('Patient', patient)).data;
```

Comment:
I think that these change will cause the search tests to eventually fail.

Tests relied on the fact that the `raceCode` was unique to ensure that the patient was part of search results. Now that the test data is always using the same `raceCode`, the query will return several results and as Patients accumulate the expected Patient will get pushed out of the first search results page and fail the test.

btw this will be a lot easier to fix if we actually supported searching by `_lastUpdated` or using `_sort` to get the newest resources first 😉

Reply:
The test was changed to used the same `raceCode` because creating an invalid `raceCode` causes validation to fail during the create step. 

```
{
    "resourceType": "OperationOutcome",
    "text": {
        "status": "generated",
        "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\"><h1>Operation Outcome</h1><table border=\"0\"><tr><td style=\"font-weight: bold;\">error</td><td>[]</td><td><pre>Patient.extension[0].extension[0].value.ofType(Coding) - Unknown code 'urn:oid:2.16.840.1.113883.6.238#abc' for 'urn:oid:2.16.840.1.113883.6.238#abc'</pre></td></tr></table></div>"
    },
    "issue": [
        {
            "severity": "error",
            "code": "invalid",
            "diagnostics": "Patient.extension[0].extension[0].value.ofType(Coding) - Unknown code 'urn:oid:2.16.840.1.113883.6.238#abc' for 'urn:oid:2.16.840.1.113883.6.238#abc'"
        }
    ]
}
```

What are your thoughts on narrowing the search result to one result by adding name as part of the search parameter? For example: 
```
p({ race: raceCode, name: testPatient.name[0].family }),
p({ ethnicity: ethnicityCode, name: testPatient.name[0].family }),
```

This isn't ideal because we're searching on more than one parameter. However, it does still confirm that searching on `race` and `ethnicityCode` works.

Reply:
that works